beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -125,19 +125,54 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 		sort.Slice(blocks, func(i, j int) bool {
 			return blocks[i].Slot < blocks[j].Slot
 		})
+		var checkParentExists func(blk *eth.BeaconBlock, parentBlocks []*eth.BeaconBlock) ([]*eth.BeaconBlock, error)
+		checkParentExists = func(blk *eth.BeaconBlock, parentBlocks []*eth.BeaconBlock) ([]*eth.BeaconBlock, error) {
+			ok, err := s.chain.ParentExists(ctx, blk)
+			if err != nil {
+				return nil, err
+			}
+			if !ok {
+				bl, err := s.requestBlocksByRoot(ctx, [][32]byte{bytesutil.ToBytes32(blk.ParentRoot)}, peers[0])
+				if err != nil {
+					return nil, err
+				}
+				parentBlocks = append(bl, parentBlocks...)
```

Comment:
you are only ever looking for 1 parent block, a slice isnt needed here

Reply:
changed